src/fragment-generation-utils.js

```diff
@@ -157,6 +159,66 @@ export const generateFragment = (selection, startTime = Date.now()) => {
   return {status: GenerateFragmentStatus.AMBIGUOUS};
 };
 
+/**
+ * Checks whether fragment generation can be attempted for a given range. This
+ * checks a handful of simple conditions: the range must be nonempty, not inside
+ * an <input>, etc. A true return is not a guarantee that fragment generation
+ * will succeed; instead, this is a way to quickly rule out generation in cases
+ * where a failure is predictable.
+ * @param {Range} range
+ * @return {boolean} - true if fragment generation may proceed; false otherwise.
+ */
+export const isValidRangeForFragmentGeneration = (range) => {
```

Comment:
Could we also validate the range's content somehow? For instance, when I select only an image, the range's textual value is some kind of white space/tab/returns combination or something, but no human readable characters.

Reply:
I think the regex check against `NON_BOUNDARY_CHARS` (line 179 below) accomplishes what you're describing.

Reply:
Gotcha!